iiif_pipeline/pipeline.py

```diff
@@ -0,0 +1,68 @@
+import logging
+import os
+import shortuuid
+from configparser import ConfigParser
+
+from .clients import ArchivesSpaceClient, AWSClient
+from .derivatives import create_jp2, create_pdf
+from .manifests import ManifestMaker
+from .helpers import matching_files
+
+
+class IIIFPipeline:
+    def __init__(self):
+        logfile = 'iiif_generation.log'
+        logging.basicConfig(filename=logfile,
+                            level=logging.INFO)
+        self.config = ConfigParser()
+        self.config.read("local_settings.cfg")
+
+    def run(self, source_dir, skip):
+        """Instantiates and runs derivative creation, manifest creation, and AWS upload files.
+
+        Args:
+            source_directory (str): Directory path to original source files.
```

Comment:
Related to the comment on the arguments the script is expecting; the source_directory variable does not actually refer to the directory where the actual tiff files are, but is a couple of directories up.